server/src/main/java/org/opensearch/client/node/NodeClient.java

```diff
@@ -57,26 +60,29 @@
  */
 public class NodeClient extends AbstractClient {
 
-    private Map<ActionType, TransportAction> actions;
+    private DynamicActionRegistry actionRegistry;
+    private ActionFilters actionFilters;
```

Comment:
`actionFilters` is not used, please remove (along with `transportService` and `extensionsManager`)

Reply:
You'd think the yellow-squigglies in my IDE would have reminded me to do that...